### ISSUE TITLE
wrong user image when using multiple accounts 

### ISSUE THREAD

**Issue by alexanderdd** [stefan-niedermann/nextcloud-deck#324]
**Describe the bug**
When you use multiple accounts, the user image from one account is used for all accounts.

**To Reproduce**
1. have multiple NC instances with different user images, 
2. connect those instances to nextcloud deck app on android
3. one user image is used for all accounts, also when assigning yourself to a card

**Expected behavior**
different user images should be respected as in Nextcloud Android app

**Smartphone (please complete the following information):**
 - Android-Version: 8.0.0
- Nextcloud Deck Android version 0.1.4
- Nextcloud Deck web version 0.8.0

**Comment by desperateCoder** [comment]
Thanks for the report @alexanderdd, I can confirm.

**Comment by stefan-niedermann** [comment]
Switching works correctly, also the correct file is requested by `Glide`. I guess, it's some kind of tricky caching-mechanism, maybe because the `target()`s are identically :thinking: 

```
Selected account : stefan@https://nextcloud.niedermann.it
Selected account : https://nextcloud.niedermann.it
Requesting URI: https://nextcloud.niedermann.it/index.php/avatar/stefan/105
Selected account : https://nextcloud.niedermann.it
Requesting URI: https://nextcloud.niedermann.it/index.php/avatar/stefan/84
Selected account : https://nextcloud.niedermann.it
Selected account : https://nextcloud.niedermann.it
Requesting URI: https://nextcloud.niedermann.it/index.php/avatar/artur/84
Selected account : https://nextcloud.niedermann.it
Selected account : https://nextcloud.niedermann.it
Selected account : https://nextcloud.niedermann.it
Requesting URI: https://nextcloud.niedermann.it/index.php/avatar/stefan/84
Selected account : https://nextcloud.niedermann.it
Selected account : https://nextcloud.niedermann.it
Requesting URI: https://nextcloud.niedermann.it/index.php/avatar/stefan/84
*** Selected account : stefan@https://nextcloud.knecht.rocks
*** Selected account : https://nextcloud.knecht.rocks
*** Requesting URI: https://nextcloud.knecht.rocks/index.php/avatar/stefan/105
*** Selected account : https://nextcloud.knecht.rocks
*** Requesting URI: https://nextcloud.knecht.rocks/index.php/avatar/stefan/84
Selected account : stefan@https://nextcloud.niedermann.it
Selected account : https://nextcloud.niedermann.it
Requesting URI: https://nextcloud.niedermann.it/index.php/avatar/stefan/105
Selected account : https://nextcloud.niedermann.it
Requesting URI: https://nextcloud.niedermann.it/index.php/avatar/stefan/84
Selected account : https://nextcloud.niedermann.it
Selected account : https://nextcloud.niedermann.it
Requesting URI: https://nextcloud.niedermann.it/index.php/avatar/artur/84
Selected account : https://nextcloud.niedermann.it
Selected account : https://nextcloud.niedermann.it
Selected account : https://nextcloud.niedermann.it
Requesting URI: https://nextcloud.niedermann.it/index.php/avatar/stefan/84
Selected account : https://nextcloud.niedermann.it
Selected account : https://nextcloud.niedermann.it
Requesting URI: https://nextcloud.niedermann.it/index.php/avatar/stefan/84

```

**Comment by stefan-niedermann** [comment]
Note to future @stefan-niedermann: I guess `Glide.with(context).clear(imageView)` might be useful though i didn't figure out yet how to exactly use it and if it can fix the issue at all.

**Comment by stefan-niedermann** [comment]
I did some further tests (branch [`324-avatars-not-updating-proof-of-concept`](https://github.com/stefan-niedermann/nextcloud-deck/compare/324-avatars-not-updating-proof-of-concept?expand=1)) and research and i think the root cause is the way we are initializing the `SingleSignOnLibraryGlideModule`. The problem is, that we are requesting one time the current `SingleSignOnAccount` and build one time a `NextcloudAPI`-object with this account.

I have made a Proof of Concept which will create a new `NextcloudAPI`-instance for each and every image request and stop and destroy it after loading the image. This is crazily bad architecture and not performant and must therefore not be merged into `master`-branch. But it proofs, that we have a problem at this point.

I see two options here:
- Use the `SingleAccountHelper.registerSharedPreferenceChangeListener`-method and recreate the `NextcloudAPI` when a new `SingleSignOnAccount` has been set
- Store the `NextcloudAPI`-objects in a Map with the account name as the key
  - This is what i am doing in the `Notes`-app
  - The [recommendation](https://github.com/nextcloud/Android-SingleSignOn/issues/120#issuecomment-540069990) is to keep the `NextcloudAPI`-instances as long open as possible
